### PR TITLE
Add SDL2+OpenGL3 backend to imgui

### LINKED ISSUE #1687: Adding more backend options for imgui
### Is your feature request related to a problem? Please describe.

I have a project that I am thinking about converting to use Xmake, and it uses the Dear ImGui library. However, it uses the SDL2+OpenGL3 backends, which I cannot seem to install through xrepo.

The imgui package offers a few set configurations, like GLFW+OpenGL3 and SDL2+SDL Renderer, but these may not work for all projects (such as mine).

### Describe the solution you'd like

The imgui package should provide additional backend config options (such as SDL2+OpenGL3).

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by AidanSun05**
This PR closes #1687.

There's some duplicate code between the GLFW+OpenGL3 and SDL2+OpenGL3 configs - I didn't refactor this because I didn't want to affect existing code. Should this be cleaned up before merging?

### CODE REVIEW

**Comment by waruqi** [comment]
> This PR closes #1687.
> 
> There's some duplicate code between the GLFW+OpenGL3 and SDL2+OpenGL3 configs - I didn't refactor this because I didn't want to affect existing code. Should this be cleaned up before merging?

you can try

File: packages/i/imgui/port/xmake.lua [hunk 1]
```diff
@@ -26,6 +27,12 @@ end
 if has_config("sdl2") then
     add_requires("libsdl >=2.0.17")
 end
+if has_config("sdl2_opengl3") then
+    add_requires("libsdl >=2.0.17")
```
Comment (waruqi) [inline packages/i/imgui/port/xmake.lua:31]:
```lua
if has_config("sdl2") or has_config("sdl2_opengl3") then
        add_requires("libsdl >=2.0.17")

```

File: packages/i/imgui/port/xmake.lua [hunk 2]
```diff
@@ -62,13 +69,25 @@ target("imgui")
...
     if has_config("sdl2") then
         add_files("backends/imgui_impl_sdl.cpp", "backends/imgui_impl_sdlrenderer.cpp")
         add_headerfiles("backends/imgui_impl_sdl.h", "backends/imgui_impl_sdlrenderer.h")
         add_packages("libsdl")
     end
 
+    if has_config("sdl2_opengl3") then
+        add_files("backends/imgui_impl_sdl.cpp", "backends/imgui_impl_opengl3.cpp")
+        add_headerfiles("backends/imgui_impl_sdl.h", "backends/imgui_impl_opengl3.h")
+        add_packages("libsdl")
```
Comment (waruqi) [inline packages/i/imgui/port/xmake.lua:82]: and here

**Comment by AidanSun05** [comment]
I pushed a commit that cleans up the duplicate code. I also tried installing imgui again with each backend option turned on separately to make sure the refactor didn't break anything.

**Pull request merged**